packages/governance/src/question.js

```diff
@@ -0,0 +1,195 @@
+// @ts-check
+
+import { Far, passStyleOf } from '@agoric/marshal';
+import { sameStructure } from '@agoric/same-structure';
+import { makeHandle } from '@agoric/zoe/src/makeHandle.js';
+import { Nat } from '@agoric/nat';
+
+import { assertType, ParamType } from './paramManager.js';
+
+const { details: X, quote: q } = assert;
+
+// Topics being voted on are 'Questions'. Before a Question is known to a
+// registrar, the parameters can be described with a QuestionSpec. Once the
+// question has been presented to a Registrar, there is a QuestionDetails record
+// that also includes the VoteCounter which will determine the outcome and the
+// questionHandle that uniquely identifies it.
+
+/**
+ * "unranked" is more formally known as "approval" voting, but this is hard for
+ * people to intuit when there are only two alternatives.
+ *
+ * @type {{
+ *   UNRANKED: 'unranked',
+ *   ORDER: 'order',
+ * }}
+ */
+const ChoiceMethod = {
+  UNRANKED: 'unranked',
+  ORDER: 'order',
+};
+
+/** @type {{
+ *   PARAM_CHANGE: 'param_change',
+ *   ELECTION: 'election',
+ *   SURVEY: 'survey',
+ * }}
+ */
+const ElectionType = {
+  // A parameter is named, and a new value proposed
+  PARAM_CHANGE: 'param_change',
+  // choose one or multiple winners, depending on ChoiceMethod
+  ELECTION: 'election',
+  SURVEY: 'survey',
+};
+
+/** @type {{
+ *   MAJORITY: 'majority',
+ *   NO_QUORUM: 'no_quorum',
+ *   ALL: 'all',
+ * }}
+ */
+const QuorumRule = {
+  MAJORITY: 'majority',
+  NO_QUORUM: 'no_quorum',
+  // The election isn't valid unless all voters vote
+  ALL: 'all',
+};
+
+/** @param {SimpleIssue} issue */
+const looksLikeSimpleIssue = issue => {
+  assert.typeof(
+    issue.text,
+    'string',
+    X`Issue ("${issue}") must be a record with text: aString`,
+  );
```

Comment:
Hm. Repeating the error message is a little awkward. But written this way, `tsc` helps us cover all the cases:

```suggestion
  assert.typeof(issue, 'object', X`Issue ("${issue}") must be a record`);
  assert(
    issue && typeof issue.text === 'string',
    X`Issue ("${issue}") must be a record with text: aString`,
  );
  return undefined;
```

Reply:
done.

This one's not as easy to search everywhere for matching patterns. If you see others, please point them out.